scripts/seed-database.ts

```diff
@@ -0,0 +1,81 @@
+import { createConnection, Connection } from 'mysql2/promise'
+import { scryptHash } from '../src/plugins/custom/scrypt.js'
+
+async function seed () {
+  const connection: Connection = await createConnection({
+    multipleStatements: true,
+    host: process.env.MYSQL_HOST,
+    port: Number(process.env.MYSQL_PORT),
+    database: process.env.MYSQL_DATABASE,
+    user: process.env.MYSQL_USER,
+    password: process.env.MYSQL_PASSWORD
+  })
+
+  try {
+    await truncateTables(connection)
+    await seedUsers(connection)
+  } catch (error) {
+    console.error('Error seeding database:', error)
+  } finally {
+    await connection.end()
+  }
+}
+
+async function truncateTables (connection: Connection) {
+  const [tables]: any[] = await connection.query('SHOW TABLES')
+
+  if (tables.length > 0) {
+    const tableNames = tables.map(
+      (row: { [key: string]: string }) => row[`Tables_in_${process.env.MYSQL_DATABASE}`]
+    )
+    const truncateQueries = tableNames
+      .map((tableName: string) => `TRUNCATE TABLE \`${tableName}\``)
+      .join('; ')
+
+    await connection.query('SET FOREIGN_KEY_CHECKS = 0')
+    try {
+      await connection.query(truncateQueries)
+      console.log('All tables have been truncated successfully.')
+    } finally {
+      await connection.query('SET FOREIGN_KEY_CHECKS = 1')
+    }
+  }
+}
+
+async function seedUsers (connection: Connection) {
+  const usernames = ['basic', 'moderator', 'admin']
+  const hash = await scryptHash('password123$')
+
+  // The goal here is to create a role hierarchy
+  // E.g. an admin should have all the roles
+  const rolesAccumulator: number[] = []
+
+  for (const username of usernames) {
+    const [userResult]: any[] = await connection.execute(`
```

Comment:
To improve type safety, instead of forcing the type via casting, can we create an use the `InsertResult` type ?

```
type InsertResult = {
  insertId: number;
}

const [userResult]: InsertResult[] = ...

const userId = (userResult)...
```

Reply:
Here, `any` really is `[QueryResult, FieldPacket[]]` so it would not work, `mysql2` typing is pretty bad, they don't offer generics if I understand correctly. Maybe we could use `knex` here too. I will create an issue about it to investigate.